[Security Solution] Actualize prebuilt rule customization test plan (#222188)
elastic/kibana: x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_customization.md

**Automation**: 1 cypress test.

```Gherkin
Given a prebuilt rule installed
And it is customized
When user changes any rule field value (so it differs from the base version) in rule edit form
Then the rule is successfully updated
And the "Modified" badge should appear on the rule's detail page
```

#### **Scenario: User can navigate to prebuilt rule editing page from the rule details page**

**Automation**: 1 cypress test.

```Gherkin
Given a prebuilt rule installed
And the user has navigated to its rule details page
When a user clicks overflow button on this rule
Then the "edit rule settings" button should be enabled
And should bring the user to the rule edit page when clicked on
```

#### **Scenario: User can navigate to prebuilt rule editing page from the rule management page**

**Automation**: 1 cypress test.

```Gherkin
Given a prebuilt rule installed
And the user has navigated to the rule management page
When a user clicks overflow button on this rule
Then the "edit rule settings" button should be enabled
And should bring the user to the prebuilt rule edit page when clicked on
```

> #### **Scenario: User can bulk edit prebuilt rules from rules management page**

Reviewer: We have separate scenarios for editing a customized and non-cust. prebuilt rules:
- User can edit a customized prebuilt rule from the rule edit page
- User can edit a non-customized prebuilt rule from the rule edit page

Do we want to treat bulk actions the same way? Like:
- User can bulk edit prebuilt rules that weren't customized
- User can bulk edit prebuilt rules that were previosly customized

wdyt?

Author: Ideally, bulk editing would require 3 scenarios to test all the cases

- bulk edit non-customized prebuilt rules
- bulk edit customized prebuilt rules
- bulk edit a mix of customized and non-customized prebuilt rules

But it's sufficient to have only the last one to reduce maintainability costs. So I've updated the test scenario accordingly.

> **Automation**: 1 integration test.

Reviewer: Should we test this with different endpoints: single update, single patch, bulk update?

> **Automation**: 1 cypress test.

Reviewer: We need to change the last "modified" to "customized" in scenario title for consistency with other scenarios.

"Modified badge should appear on the rule management table when prebuilt rule is ~~modified~~ **customized**"

**Automation**: 1 cypress test.

```Gherkin
Given a prebuilt rule installed
And that rule is customized
And that rule has an upgrade
When a user navigates to the rule updates table
And the "Modified" badge should be present in the table row
```

#### **Scenario: Modified badge should not appear on the rule updates table when prebuilt rule isn't customized**

**Automation**: 1 cypress test.

```Gherkin
Given a prebuilt rule installed
And that rule is non-customized
And that rules has an upgrade
When a user navigates to the rule updates table
And the "Modified" badge should NOT be present in the table row
```

> #### **Scenario: User should be able to see only customized rules in the rule updates table**

Reviewer: ```suggestion
#### **Scenario: User should be able to filter by customized rules in the rule updates table**
```
For consistency with scenario below

```Gherkin
Given a Kibana instance running under an insufficient license
And a prebuilt rule installed
When a user selects one prebuilt rule in the rule management table
And user also selects one custom rule
And user attempts to apply a <customizing_bulk_action> bulk action to the selected rule
Then the user should see a message that this action is not allowed for prebuilt rules under the current license
And the required license name should be included in the message
And a button to proceed with applying the action only to custom rules should be displayed
```

> #### **Scenario: User can't edit prebuilt rules via bulk edit API under an insufficient license**

Reviewer: Do we then also want an integration test for a single rule edit? Or do we consider it covered by a Cypress test?
